src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c

```diff
@@ -3019,41 +3022,99 @@ static ssize_t program_vmr_store(struct device *dev,
 }
 static DEVICE_ATTR_WO(program_vmr);
 
-static ssize_t vmr_system_dtb_read(struct file *filp, struct kobject *kobj,
-	struct bin_attribute *attr, char *buf, loff_t off, size_t count)
+static ssize_t vmr_default_fpt_read(struct file *filp, struct kobject *kobj,
+        struct bin_attribute *attr, char *buf, loff_t off, size_t count)
 {
 	struct xocl_xgq_vmr *xgq =
-		dev_get_drvdata(container_of(kobj, struct device, kobj));
-	char *blob = NULL;
-	size_t size = 0;
-	ssize_t ret = 0;
+                dev_get_drvdata(container_of(kobj, struct device, kobj));
+        char *log_buf = NULL;
+        size_t log_size = 0;
+        ssize_t ret = 0;
 
-	/*
-	 * refresh cached data only if off is 0,
-	 * which indicates that there is a new request.
-	 */
-	if (off == 0)
-		ret = xgq_refresh_system_dtb(xgq);
+	if (off > 0)
```

Comment:
this looks incorrect. why off cannot be bigger than 0? It might be more than 4096 per linux sysfs request for the blob data. we are also able to return data with any offset.

Reply:
non zero offset means there is some date returned per read request. And this is a more like a boundary check to avoid infinite looping of this API call because after read and return, the offset seeks back to zero and automatically keeps calling this API by behavior of sysfs.